ts/src/base/Exchange.ts

```diff
@@ -6358,6 +6358,29 @@ export default class Exchange {
         return this.filterBySymbolSinceLimit (sorted, symbol, since, limit) as LongShortRatio[];
     }
 
+    handleTriggerDirectionAndParams (params, exchangeSpecificKey: string) {
+        /**
+         * @ignore
+         * @method
+         * @returns {[string, object]} the trigger-direction value and omited params
+         */
+        let triggerDirection = this.safeString (params, 'triggerDirection');
+        // throw exception if:
+        // A) if provided value is not unified (support old "up/down" strings too)
+        // B) if exchange specific "trigger direction key" (eg. "stopPriceSide") was not provided
+        if (!this.inArray (triggerDirection, [ 'ascending', 'descending', 'up', 'down', 'above', 'below' ]) && !(exchangeSpecificKey in params)) {
+            throw new ArgumentsRequired (this.id + ' createOrder() : trigger orders require params["triggerDirection"] to be either "ascending" or "descending"');
+        }
+        // if old format was provided, overwrite to new
+        if (triggerDirection === 'up' || triggerDirection === 'above') {
+            triggerDirection = 'ascending';
+        } else if (triggerDirection === 'down' || triggerDirection === 'below') {
+            triggerDirection = 'descending';
+        }
+        params = this.omit (params, [ 'ascending', 'descending' ]);
+        return [ triggerDirection, params ];
```

Comment:
I see a couple of issues here:
- first, if the user provides the exchange-specific value you're going to return `undefined`. is this expected?
- second, why omit `ascending/descending` if that's the value but not the key (triggerDirection)?

Reply:
https://discord.com/channels/690203284119617602/1317080820934246421/1318640217074307152